src/Control/ElementFormController.php

```diff
@@ -55,7 +56,11 @@ public function finished()
         $user->finished();
 
         $page = $this->getPage();
-        $controller = Injector::inst()->create($page->getControllerName(), $this->element->getPage());
+
+	while(!$page instanceof SiteTree) {
```

Comment:
Identation is misaligned

Next extra brackets when negating instanceof e.g. `(!($page instanceof SiteTree))`

Need to check for presence of `null` since that's a valid return type for BaseElement::getPage() - if null then need to handle that otherwise will get stuck in infinite while loop

Reply:
Hi, sure, have added condition for null, thanks